### PR TITLE
added rpi version of rev4

### PR DESCRIPTION
**Pull request by Rossman360**
## Description

As best I understand, because my controller uses a different SPI flash than some of the more standard ones, I cannot use a converter and thus need this separate rev.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/montsinger/rebound/rev4pi/rules.mk [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# MCU name
+MCU = RP2040
+BOOTLOADER = rp2040
+BOARD = GENERIC_RP_RP2040
+
+
```
Comment (waffle87) [inline keyboards/montsinger/rebound/rev4pi/rules.mk:6]:
```suggestion
```

File: keyboards/montsinger/rebound/rev4pi/info.json [hunk 2]
```diff
@@ -0,0 +1,132 @@
+{
+    "keyboard":"Rebound Rev4pi",
+    "manufacturer": "Montsinger",
+    "url": "https://montsinger.net",
+    "maintainer": "rossman360",
+    "usb": {
+        "vid": "0x524D",
+        "pid": "0x0001",
+        "device_version": "0.0.2"
+    },
```
Comment (waffle87) [inline keyboards/montsinger/rebound/rev4pi/info.json:10]:
```suggestion
    },
    "processor": "RP2040",
    "bootloader": "rp2040",
    "board": "GENERIC_RP_RP2040",
```

File: keyboards/montsinger/rebound/rev4pi/rules.mk [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+
+RGB_MATRIX_ENABLE = no
+RGB_MATRIX_DRIVER = WS2812
+
+WS2812_DRIVER = vendor
+ENCODER_ENABLE = yes
```
Comment (waffle87) [inline keyboards/montsinger/rebound/rev4pi/rules.mk:24]: If encoders are enabled at the keyboard level, there should be keyboard level encoder functionality as well, that way it works for QMK Configurator users.
eg. in a `rev4pi.c` file using `encoder_update_kb()` function.

**Review by waffle87 — CHANGES_REQUESTED** [review]
There are also dependency commits that need to be reverted.

File: keyboards/montsinger/rebound/rev4pi/keymaps/rossman360/keymap.c [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+ case _FN1:
+    if (clockwise) {
+      tap_code16(S(KC_TAB));
+    } else {
+      tap_code16(KC_TAB);
+    }
+   break;
+}
+    return true;
+}
```
[the remark this answers is not in the document]
Comment (Rossman360) [reply keyboards/montsinger/rebound/rev4pi/keymaps/rossman360/keymap.c]: i just deleted all the layered ones as I never use them

File: keyboards/montsinger/rebound/rev4pi/rules.mk (hunk 3 above)
Comment (Rossman360) [reply keyboards/montsinger/rebound/rev4pi/rules.mk:22]: I need a little help understanding this one ... can't find any examples.

**Comment by Rossman360** [comment]
I nuked everything and started over. Everything seems to be working now and all request changes have been implemented except for encoder_update_kb(), which I could use a little more guidance on.

File: keyboards/montsinger/rebound/rev4pi/rules.mk (hunk 3 above)
Comment (waffle87) [reply keyboards/montsinger/rebound/rev4pi/rules.mk:22]: (replied on discord)

File: keyboards/montsinger/rebound/rev4pi/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,103 @@
+#include QMK_KEYBOARD_H
```
Comment (waffle87) [inline keyboards/montsinger/rebound/rev4pi/keymaps/default/keymap.c:1]: This file is missing a license header.

File: keyboards/montsinger/rebound/rev4pi/keymaps/rossman360/keymap.c [hunk 6]
```diff
@@ -0,0 +1,94 @@
+#include QMK_KEYBOARD_H
```
Comment (waffle87) [inline keyboards/montsinger/rebound/rev4pi/keymaps/rossman360/keymap.c:1]: This file is missing a license header.

File: keyboards/montsinger/rebound/rev4pi/rules.mk [hunk 7]
```diff
@@ -0,0 +1,25 @@
+# MCU name
+MCU = RP2040
+BOOTLOADER = rp2040
+BOARD = GENERIC_RP_RP2040
```
Comment (waffle87) [inline keyboards/montsinger/rebound/rev4pi/rules.mk:4]:
```suggestion
```

File: keyboards/montsinger/rebound/rev4pi/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (clockwise) {
+      tap_code16(S(KC_VOLD));
+    } else {
+      tap_code16(KC_VOLU);
+    }
+    return true;
+}
```
Comment (waffle87) [inline keyboards/montsinger/rebound/rev4pi/keymaps/default/keymap.c:103]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/montsinger/rebound/rev4pi/config.h [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+#define DEBOUNCE 5
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* Bootmagic Lite key configuration */
+#define BOOTMAGIC_LITE_ROW 0
+#define BOOTMAGIC_LITE_COLUMN 0
```
[the remark this answers is not in the document]
Comment (Rossman360) [reply keyboards/montsinger/rebound/rev4pi/config.h]: neat

File: keyboards/montsinger/rebound/rev4pi/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+        case DVORAK:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_DVORAK);
+            }
+            break;
+    }
+    return true;
+}
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
```
Comment (drashna) [inline keyboards/montsinger/rebound/rev4pi/keymaps/via/keymap.c:112]: Should probably use encoder map here.

Comment (Rossman360) [reply keyboards/montsinger/rebound/rev4pi/keymaps/via/keymap.c]: deleted. it's already in the .c file

**Review by drashna — APPROVED** [review]

**Comment by Rossman360** [comment]
bump

File: keyboards/montsinger/rebound/rev4pi/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+};
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+  COLEMAK,
+  DVORAK
+};
+
+#define LOWER  MO(_LOWER)
+#define RAISE  MO(_RAISE)
```
Comment (waffle87) [inline keyboards/montsinger/rebound/rev4pi/keymaps/via/keymap.c:40]:
```suggestion
```

File: keyboards/montsinger/rebound/rev4pi/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+  KC_ESC,  KC_A,    KC_R,    KC_S,    KC_T,    KC_D,   _______,  KC_H,    KC_N,    KC_E,    KC_I,    KC_O,    KC_QUOT,
+  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,   _______,  KC_K,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT ,
+  ADJUST,  KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC, _______,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
+),
+
+[_DVORAK] = LAYOUT_all(
+  KC_TAB,  KC_QUOT, KC_COMM, KC_DOT,  KC_P,    KC_Y,             KC_F,    KC_G,    KC_C,    KC_R,    KC_L,    KC_BSPC,
+  KC_ESC,  KC_A,    KC_O,    KC_E,    KC_U,    KC_I,   _______,  KC_D,    KC_H,    KC_T,    KC_N,    KC_S,    KC_SLSH,
+  KC_LSFT, KC_SCLN, KC_Q,    KC_J,    KC_K,    KC_X,   _______,  KC_B,    KC_M,    KC_W,    KC_V,    KC_Z,    KC_ENT ,
+  ADJUST,  KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC, _______,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```
Comment (waffle87) [inline keyboards/montsinger/rebound/rev4pi/keymaps/via/keymap.c:63]:
```suggestion
   ADJUST,  KC_LCTL, KC_LALT, KC_LGUI, TL_LOWR,   KC_SPC, KC_ENT,  KC_SPC,  TL_UPPR,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
),

[_COLEMAK] = LAYOUT_all(
  KC_TAB,  KC_Q,    KC_W,    KC_F,    KC_P,    KC_G,             KC_J,    KC_L,    KC_U,    KC_Y,    KC_SCLN, KC_BSPC,
  KC_ESC,  KC_A,    KC_R,    KC_S,    KC_T,    KC_D,   _______,  KC_H,    KC_N,    KC_E,    KC_I,    KC_O,    KC_QUOT,
  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,   _______,  KC_K,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT ,
  ADJUST,  KC_LCTL, KC_LALT, KC_LGUI, TL_LOWR,   KC_SPC, _______,  KC_SPC,  TL_UPPR,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
),

[_DVORAK] = LAYOUT_all(
  KC_TAB,  KC_QUOT, KC_COMM, KC_DOT,  KC_P,    KC_Y,             KC_F,    KC_G,    KC_C,    KC_R,    KC_L,    KC_BSPC,
  KC_ESC,  KC_A,    KC_O,    KC_E,    KC_U,    KC_I,   _______,  KC_D,    KC_H,    KC_T,    KC_N,    KC_S,    KC_SLSH,
  KC_LSFT, KC_SCLN, KC_Q,    KC_J,    KC_K,    KC_X,   _______,  KC_B,    KC_M,    KC_W,    KC_V,    KC_Z,    KC_ENT ,
  ADJUST,  KC_LCTL, KC_LALT, KC_LGUI, TL_LOWR,   KC_SPC, _______,  KC_SPC,  TL_UPPR,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```

File: keyboards/montsinger/rebound/rev4pi/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+  _______, QK_BOOT,   _______, _______, _______, _______,          _______, _______, _______, _______, _______, KC_DEL,
+  _______, _______, _______, AU_ON,   AU_OFF,  AG_NORM, _______, AG_SWAP, QWERTY,  COLEMAK, DVORAK,  _______, _______,
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+)
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
```
Comment (waffle87) [inline keyboards/montsinger/rebound/rev4pi/keymaps/via/keymap.c:90]:
```suggestion
```

File: keyboards/montsinger/rebound/rev4pi/rules.mk [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+
```
Comment (waffle87) [inline keyboards/montsinger/rebound/rev4pi/rules.mk:13]:
```suggestion
TRI_LAYER_ENABLE = yes

```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by Rossman360** [comment]
that should be erryting, thanks waffle

File: keyboards/montsinger/rebound/rev4pi/info.json [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+    "processor": "RP2040",
```
Comment (fauxpark) [inline keyboards/montsinger/rebound/rev4pi/info.json:11]:
```suggestion
    "processor": "RP2040",
    "encoder": {
        "rotary": [
            {"pin_a": "GP0", "pin_b": "GP1"}
        ]
    },
```

File: keyboards/montsinger/rebound/rev4pi/config.h [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+*/
+
+#pragma once
+
+#define MATRIX_ROW_PINS { GP2, GP3, GP4, GP5, GP26, GP27, GP28, GP29 }
+#define MATRIX_COL_PINS { GP6, GP7, GP8, GP9, GP23, GP20, GP22 }
+
+#define ENCODERS_PAD_A { GP0 }
+#define ENCODERS_PAD_B { GP1 }
+
```
Comment (fauxpark) [inline keyboards/montsinger/rebound/rev4pi/config.h:22]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by Rossman360** [comment]
boom, roasted

File: keyboards/montsinger/rebound/rev4pi/config.h [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#define MATRIX_ROW_PINS { GP2, GP3, GP4, GP5, GP26, GP27, GP28, GP29 }
+#define MATRIX_COL_PINS { GP6, GP7, GP8, GP9, GP23, GP20, GP22 }
+
```
Comment (waffle87) [inline keyboards/montsinger/rebound/rev4pi/config.h:19]: This also needs to be migrated to `info.json` now.

File: keyboards/montsinger/rebound/rev4pi/rev4pi.h [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+    ) { \
+  { K00, K01, K02, KC_NO, K04, K05, K06}, \
+  { K10, K11, K12, K13, K14, K15, K16}, \
+  { K20, K21, K22, K23, K24, K25, K26}, \
+  { K30, K31, K32, K33, K34, K35, K36}, \
+  { K40, K41, K42, KC_NO, K44, K45, K46}, \
+  { K50, K51, K52, KC_NO, K54, K55, K56}, \
+  { K60, K61, K62, KC_NO, K64, K65, K66}, \
+  { K70, K71, K72, KC_NO, K74, K75, K76} \
+}
```
Comment (waffle87) [inline keyboards/montsinger/rebound/rev4pi/rev4pi.h:58]: And this. You can use `qmk info -f json -kb montsinger/rebound/rev4pi` to help. Then this file can be removed.

File: keyboards/montsinger/rebound/rev4pi/rev4pi.c [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "rev4pi.h"
```
Comment (waffle87) [inline keyboards/montsinger/rebound/rev4pi/rev4pi.c:16]:
```suggestion
#include "quantum.h"
```

**Review by waffle87 — COMMENTED** [review]
Soz this got left behind.

File: keyboards/montsinger/rebound/rev4pi/rev4pi.h (hunk 18 above)
Comment (Rossman360) [reply keyboards/montsinger/rebound/rev4pi/rev4pi.h:58]: removing this code code causes this error when compiling: 

./keyboards/montsinger/rebound/rev4pi/keymaps/rossman360/keymap.c:36:11: error: implicit declaration of function 'LAYOUT_all' [-Werror=implicit-function-declaration]
   36 | [_BASE] = LAYOUT_all(
      |           ^~~~~~~~~~
./keyboards/montsinger/rebound/rev4pi/keymaps/rossman360/keymap.c:36:11: error: initializer element is not constant
./keyboards/montsinger/rebound/rev4pi/keymaps/rossman360/keymap.c:36:11: note: (near initialization for 'keymaps[0][0][0]')
In file included from quantum/keymap_introspection.c:5:

(for each layer in my keymap)

Comment (Rossman360) [reply keyboards/montsinger/rebound/rev4pi/rev4pi.h]: okay I dunno what I did differently this time but it seems to be compiling now

Comment (Rossman360) [reply keyboards/montsinger/rebound/rev4pi/rev4pi.h]: oh, it's worth noting that deleting all the code from rev4pi.h works fine, but removing the file itself returns the error:

keyboards/montsinger/rebound/rebound.h:30:14: fatal error: rev4pi.h: No such file or directory
   30 |     #include "rev4pi.h"
      |              ^~~~~~~~~~
compilation terminated.

File: keyboards/montsinger/rebound/rev4pi/info.json [hunk 20]
```diff
@@ -0,0 +1,193 @@
...
+        "extrakey": true,
+        "mousekey": false,
+        "nkro": false,
+        "oled": false,
+        "rgb_matrix": false,
+        "rgblight": false
+    },
+    "keyboard": "Rebound Rev4pi",
+    "keyboard_folder": "montsinger/rebound/rev4pi",
+    "keymaps": {},
```
Comment (fauxpark) [inline keyboards/montsinger/rebound/rev4pi/info.json:48]:
```suggestion
```

File: keyboards/montsinger/rebound/rev4pi/info.json [hunk 21]
```diff
@@ -0,0 +1,193 @@
...
+                { "label": "K44", "matrix": [4, 4], "w": 1, "x": 8, "y": 3 },
+                { "label": "K35", "matrix": [3, 5], "w": 1, "x": 9, "y": 3 },
+                { "label": "K45", "matrix": [4, 5], "w": 1, "x": 10, "y": 3 },
+                { "label": "K36", "matrix": [3, 6], "w": 1, "x": 11, "y": 3 },
+                { "label": "K46", "matrix": [4, 6], "w": 1, "x": 12, "y": 3 }
+            ]
+        },
+        "LAYOUT_ortho_4x12": {
+            "c_macro": false,
+            "filename": "keyboards/montsinger/rebound/rev4pi/rev4pi.h",
```
Comment (fauxpark) [inline keyboards/montsinger/rebound/rev4pi/info.json:140]:
```suggestion
```

File: keyboards/montsinger/rebound/rev4pi/rev4pi.h [hunk 22]
```diff
@@ -0,0 +1,28 @@
+/* Copyright 2023 Ross Montsinger (@Rossman360)
```
Comment (fauxpark) [inline keyboards/montsinger/rebound/rev4pi/rev4pi.h:1]: This file can be deleted.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by Rossman360** [comment]
thanks, i fink i did it all

File: keyboards/montsinger/rebound/rebound.h [hunk 23]
```diff
@@ -1,4 +1,4 @@
-/* Copyright 2020 Ross Montsinger
+/* Copyright 2023 Ross Montsinger
```
Comment (fauxpark) [inline keyboards/montsinger/rebound/rebound.h:1]:
```suggestion
/* Copyright 2020 Ross Montsinger
```
This should fix the conflict. This file does not exist anymore.

File: keyboards/montsinger/rebound/rev4pi/keymaps/ericgebhart/keymap.c [hunk 24]
```diff
@@ -0,0 +1,101 @@
+/*
```
Comment (fauxpark) [inline keyboards/montsinger/rebound/rev4pi/keymaps/ericgebhart/keymap.c:1]: User keymaps don't need to be copied.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by fauxpark** [comment]
Perhaps it would be better to instead further split `rev4` into `rev4/base` (default) and `rev4/rp2040`.

File: keyboards/montsinger/rebound/rev4pi/keymaps/ericgebhart/keymap.c (hunk 24 above)
Comment (Rossman360) [reply keyboards/montsinger/rebound/rev4pi/keymaps/ericgebhart/keymap.c:1]: hey that's my roommate now

**Comment by Rossman360** [comment]
I figure it's simpler and makes just as much sense if I rename it to rev5

File: keyboards/montsinger/rebound/rev5/info.json [hunk 25]
```diff
@@ -0,0 +1,160 @@
+{
+    "manufacturer": "Montsinger",
+    "keyboard_name": "montsinger/rebound/rev5",
+    "maintainer": "rossman360",
+    "board": "GENERIC_RP_RP2040",
+    "bootloader": "rp2040",
+    "diode_direction": "ROW2COL",
+    "encoder": {
+        "enabled": true,
```
Comment (drashna) [inline keyboards/montsinger/rebound/rev5/info.json:9]:
```suggestion
```

File: keyboards/montsinger/rebound/rev5/info.json [hunk 26]
```diff
@@ -0,0 +1,160 @@
...
+        "enabled": false
+    },
+    "processor": "RP2040",
+    "qmk": {
+        "tap_keycode_delay": 10
+    },
+    "rgb_matrix": {
+        "driver": "WS2812"
+    },
+    "rgblight": {
```
Comment (drashna) [inline keyboards/montsinger/rebound/rev5/info.json:42]:
```suggestion
    "ws2812": {
```

File: keyboards/montsinger/rebound/rev5/keymaps/rossman360/keymap.c [hunk 27]
```diff
@@ -0,0 +1,108 @@
...
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/montsinger/rebound/rev5/keymaps/rossman360/keymap.c:15]:
```suggestion
 */
 #include QMK_KEYBOARD_H
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/montsinger/rebound/rev5/rev5.c
Comment (fauxpark) [inline keyboards/montsinger/rebound/rev5/rev5.c:1]: This file can be deleted - default encoder behaviour in core is volume control.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/montsinger/rebound/rev5/keymaps/default/keymap.c [hunk 28]
```diff
@@ -0,0 +1,99 @@
...
+enum layer_names {
+  _QWERTY,
+  _COLEMAK,
+  _DVORAK,
+  _LOWER,
+  _RAISE,
+  _ADJUST,
+};
+
+#define ADJUST MO(_ADJUST)
```
Comment (drashna) [inline keyboards/montsinger/rebound/rev5/keymaps/default/keymap.c:33]:
```suggestion
enum custom_keycodes {
    QWERTY = QK_USER,
    COLEMAK,
    DVORAK,
};

#define ADJUST MO(_ADJUST)
```

That or you eed to remove the custom keycodes.

File: keyboards/montsinger/rebound/rev5/keymaps/via/rules.mk [hunk 29]
```diff
@@ -0,0 +1,3 @@
+VIA_ENABLE = yes
+LTO_ENABLE = yes
+RGBLIGHT_ENABLE = yes
```
Comment (drashna) [inline keyboards/montsinger/rebound/rev5/keymaps/via/rules.mk:3]: Number of LEDs isn't defined, so this causes it to error out.

Number of LEDs needs to be defined, or this line remove.

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/montsinger/rebound/rev5/keymaps/via/keymap.c [hunk 30]
```diff
@@ -0,0 +1,105 @@
...
+),
+
+[_LOWER] = LAYOUT_all(
+  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC,          KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,
+  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   _______, KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE,
+  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  _______, KC_F12,S(KC_NUHS),S(KC_NUBS),_______, _______, _______,
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
+),
+
+[_RAISE] = LAYOUT_all(
```
Comment (drashna) [inline keyboards/montsinger/rebound/rev5/keymaps/via/keymap.c:71]: Too many layers.  Default is 4.  If you need more than that, you need to specify the number of layers fo dynamic keymaps.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/montsinger/rebound/rev5/info.json [hunk 31]
```diff
@@ -0,0 +1,159 @@
...
+    },
+    "mouse_key": {
+        "enabled": false
+    },
+    "processor": "RP2040",
+    "qmk": {
+        "tap_keycode_delay": 10
+    },
+    "rgb_matrix": {
+        "driver": "WS2812"
```
Comment (fauxpark) [inline keyboards/montsinger/rebound/rev5/info.json:39]: If RGB Matrix is enabled this board will not compile; it's missing the LED count define in config.h and the LED config object here.

File: keyboards/montsinger/rebound/rev5/config.h [hunk 32]
```diff
@@ -0,0 +1,29 @@
...
+*/
+
+#pragma once
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+#define RGBLED_NUM 12
```
Comment (fauxpark) [inline keyboards/montsinger/rebound/rev5/config.h:22]: This should be moved to info.json as `rgblight.led_count` if RGBLight is possible on this board.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**